pyramid/scripts/pshell.py

```diff
@@ -159,18 +162,33 @@ def run(self, shell=None):
             closer()
 
     def make_shell(self):
+        shells = {}
+
+        priority_order = ['ipython', 'bpython']
+
+        for ep in self.pkg_resources.iter_entry_points('pyramid.pshell'):
+            name = ep.name
+            shell_module = ep.load()
+            shells[name] = shell_module
+
+        sorted_shells = sorted(
+            shells.items(), key=lambda x: priority_order.index(x[0])
+        )
+
         shell = None
         user_shell = self.options.python_shell.lower()
+
         if not user_shell:
-            shell = self.make_ipython_shell()
-            if shell is None:
-                shell = self.make_bpython_shell()
+            for name, factory in sorted_shells:
+                shell = factory()
 
-        elif user_shell == 'ipython':
-            shell = self.make_ipython_shell()
+                if shell is not None:
+                    break
+        else:
+            factory = shells.get(user_shell)
 
-        elif user_shell == 'bpython':
-            shell = self.make_bpython_shell()
+            if factory is not None:
+                shell = factory()
```

Comment:
I think just add `else: raise ValueError('could not find a shell named ...')` or something to that effect to avoid fallbacks when the user explicitly requested a specific shell.